src/listener.ts

```diff
@@ -0,0 +1,24 @@
+import { NestFactory } from '@nestjs/core';
+import { MicroserviceOptions, Transport } from '@nestjs/microservices';
+import { AppModule } from './app.module';
+
+/**
+ * It creates a microservice that listens to a RabbitMQ queue and uses the AppModule to handle incoming messages
+ */
+async function bootstrap() {
+  const app = await NestFactory.createMicroservice<MicroserviceOptions>(
+    AppModule,
+    {
+      transport: Transport.RMQ,
+      options: {
+        urls: [process.env.RABBITMQ_URL],
```

Comment:
should we not use configService here instead of process.env for consistency... or what do u feel

Reply:
The two ways I know of using config service is either as a parameter in useFactory or initialize it in constructor. Neither works here, This file is akin to main.ts file but for listening. If you know a way of using config service in main.ts let me know because they are similar

Reply:
just a thought.... it is fine.. me no know sha